src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1047,3 +1052,48 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
     // activate the ramping in scratchProcess()
     m_ramp[deck] = true;
 }
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const ControllerScriptInterfaceLegacy::WellKnownCharsets targetCharset,
+        const QString& value) {
+    switch (targetCharset) {
+    case WellKnownCharsets::Latin1:
+    case WellKnownCharsets::ISO_8859_1:
+        return convertCharset(QStringLiteral("ISO-8859-1"), value);
+    case WellKnownCharsets::Latin9:
+    case WellKnownCharsets::ISO_8859_15:
+        return convertCharset(QStringLiteral("ISO-8859-15"), value);
+    case WellKnownCharsets::UCS2:
+    case WellKnownCharsets::ISO_10646_UCS_2:
+        return convertCharset(QStringLiteral("ISO-10646-UCS-2"), value);
+    default:
+        m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unknown charset specified"));
+        return QByteArray();
+    }
+}
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const QString& targetCharset, const QString& value) {
+    QByteArray encoderNameArray = targetCharset.toUtf8();
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+    auto* pCodec = QTextCodec::codecForName(encoderNameArray);
+    if (!pCodec) {
+        m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unable to open encoder"));
+        return QByteArray();
+    }
+    return pCodec->makeEncoder(QTextCodec::Flag::ConvertInvalidToNull)->fromUnicode(value);
```

Comment:
This leaks. 
> Creates a [QTextEncoder](https://doc.qt.io/qt-6/qtextencoder.html) with a specified flags to encode chunks of Unicode data as char * data.
>
> The caller is responsible for deleting the returned object.

Reply:
I think the easiest solution here would be
```suggestion
    return std::unique_ptr(pCodec->makeEncoder(QTextCodec::Flag::ConvertInvalidToNull))->fromUnicode(value);
```